cdc/server/server.go

```diff
@@ -249,17 +255,24 @@ func (s *server) etcdHealthChecker(ctx context.Context) error {
 		case <-ctx.Done():
 			return ctx.Err()
 		case <-ticker.C:
-			for _, pdEndpoint := range s.pdEndpoints {
+			endpoints, err := pc.CollectMemberEndpoints(ctx)
+			if err != nil {
+				log.Warn("etcd health check: cannot collect all members", zap.Error(err))
+				continue
+			}
+			for _, endpoint := range endpoints {
 				start := time.Now()
-				ctx, cancel := context.WithTimeout(ctx, time.Second*10)
-				resp, err := httpCli.Get(ctx, fmt.Sprintf("%s/pd/api/v1/health", pdEndpoint))
-				if err != nil {
-					log.Warn("etcd health check error", zap.Error(err))
-				} else {
-					_, _ = io.Copy(io.Discard, resp.Body)
-					_ = resp.Body.Close()
-					metrics[pdEndpoint].Observe(float64(time.Since(start)) / float64(time.Second))
+				ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+				if err := pc.Healthy(ctx, endpoint); err != nil {
+					log.Warn("etcd health check error",
+						zap.String("endpoint", endpoint), zap.Error(err))
+				}
+				metric, ok := metrics[endpoint]
+				if !ok {
+					metric = etcdHealthCheckDuration.WithLabelValues(endpoint)
+					metrics[endpoint] = metric
 				}
```

Comment:
If we cannot ensure retrieve the observer all the time, it's better to delete the `metrics` map and always use `etcdHealthCheckDuration.WithLabelValues(pdEndpoint).Observe(time.Since(start).Seconds())` to make code more clean.